mkosi/__init__.py

```diff
@@ -6285,6 +6298,49 @@ def run_ssh(args: CommandLineArguments) -> None:
         )
 
 
+def generate_secure_boot_key(args: CommandLineArguments) -> NoReturn:
+    """Generate secure boot keys using openssl"""
+    args.secure_boot_key = args.secure_boot_key or "./mkosi.secure-boot.key"
+    args.secure_boot_certificate = args.secure_boot_certificate or "./mkosi.secure-boot.crt"
```

Comment:
Do these end up in the output directory if it is defined?

Reply:
Good question. I forgot to test that. The question is: should they? I think they should end up such, that mkosi can pick them up right away without setting `--secure-boot-{key,certificate}`.

Reply:
I think they should, we do it for all the other stuff so I don't see a big reason to deviate from that here.

Reply:
Ok, I'll have a look at it tomorrow.

Reply:
So currently this would ignore `args.output_dir`, but to answer the question below, there they wouldn't be picked up automatically, since `find_secure_boot` looks relative to `args.directory`.

I'm still not sure whether supporting whether moving this relative to the output directory makes sense. `--secure-boot-key` and `--secure-boot-certificate` take paths that are not necessarily relative to the output directory. If somebody would set `--output-dir` and a relative path for either of the secure boot options the result could be confusing. 

I'm not sure what the best option is, but I think it would be great if they would be just be picked up for `--secure-boot`, which should usually be the case, if people just run the command at the root of their config. What do you think?

Reply:
We can do it this way, thinking further, these keys are not something you'll be regenerating again and again anyway so it kinda makes sense that we just output them in the working directory instead of the output dir.